library/src/webapp/bundled-js/sakai.morpheus.session.js

```diff
@@ -36,6 +36,39 @@ var setup_timeout_config = function(){
 
 }
 
+function createDHTMLMask(callback) {
+  const portalMask = document.createElement('div');
+  portalMask.id = 'portalMask';
+  portalMask.style.height = `${document.documentElement.scrollHeight}px`;
+  portalMask.style.width = '100%';
+  portalMask.style.zIndex = '1300';
+  portalMask.style.top = '0';
+  portalMask.style.left = '0';
+  portalMask.style.position = 'absolute';
+  portalMask.style.backgroundColor = 'rgba(0, 0, 0, 0.5)';
+  portalMask.style.display = 'flex';
+  portalMask.style.alignItems = 'center';
+  portalMask.style.justifyContent = 'center';
+
+  portalMask.addEventListener('click', function (event) {
+    closeDrawer();
+    event.stopPropagation();
```

Comment:
I usually stop the propagation immediately. It probably doesn't matter that much here as closeDrawer will be in the same thread and will block any further propagation while it executes.